### PR TITLE
test: add tests for HTTPSCreds

### PR DESCRIPTION
**Pull request by crenshaw-dev**
Noticed testing was light on the creds file.

### CODE REVIEW

**Review by chetan-rns — APPROVED** [review]
LGTM!

File: util/git/creds_test.go [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+			break
+		}
+	}
+	assert.Contains(t, store.creds, nonce)
+	io.Close(closer)
+	assert.NotContains(t, store.creds, nonce)
+}
+
+func TestHTTPSCreds_Environ_insecure_true(t *testing.T) {
+	store := &memoryCredsStore{creds: map[string]cred{}}
```
Comment (rishabh625) [inline util/git/creds_test.go:54]: m unsure, but shouldn't there be make used?

File: util/git/creds_test.go [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+	"strings"
+	"testing"
+)
+
+type cred struct {
+	username string
+	password string
+}
+
+type memoryCredsStore struct {
```
Comment (rishabh625) [inline util/git/creds_test.go:19]: does NoopCredsStore test should also be there?

**Review by rishabh625 — COMMENTED** [review]
Overall looks great!!

File: util/git/creds_test.go [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+	}
+	assert.False(t, found)
+}
+
+func TestHTTPSCreds_Environ_clientCert(t *testing.T) {
+	store := &memoryCredsStore{creds: map[string]cred{}}
+	creds := NewHTTPSCreds("", "", "clientCertData", "clientCertKey", false, "", store)
+	closer, env, err := creds.Environ()
+	require.NoError(t, err)
+	cert := ""
```
[the remark this answers is not in the document]
Comment (crenshaw-dev) [reply util/git/creds_test.go:94]: True, the assignment isn't necessary. Chaning!

File: util/git/creds_test.go (hunk 1 above)
Comment (crenshaw-dev) [reply util/git/creds_test.go:54]: Googled around, looks like it's up to preference. Switching to `make`.

File: util/git/creds_test.go (hunk 2 above)
Comment (crenshaw-dev) [reply util/git/creds_test.go:19]: Good point. I'll change a couple of the tests to use `NoopCredsStore`. Some of the tests don't check the creds store cleanup, so it doesn't make sense to use the memory creds store.

**Review by alexmt — APPROVED** [review]
Thank you!

**Pull request merged**